### DISCUSSION TITLE
Meson doesn't find CUDA compiler

### DISCUSSION THREAD

**Discussion by ghost** [mesonbuild/meson#12589] [Q&A] [answered]
I am trying to build the example Nvidia DOCA apps (https://docs.nvidia.com/doca/sdk/applications-overview/index.html)
However, running meson build gives the error: Could not find suitable CUDA compiler: "nvcc".
This is quite strange as I should have everything set up correctly. 
Any help/ideas would be greatly appreciated.

**Comment by dcbaker** [comment]
I'm assuming you're on Linux. is `nvcc` in your $PATH? what does the shell command `command -v nvcc` return?

**Reply by ghost** [reply]
Thanks for the reply. Yeah, I'm on Ubuntu. command -v nvcc returns /usr/local/cuda-12.2/bin/nvcc and nvcc is in my path like so: /usr/local/cuda-12.2/bin

I can compile and run CUDA applications (without Meson). After pointing to the nvcc manually in the meson.build file, it now finds it. However it gives an error "No host machine compiler for .../.../receive_tcp.cu"

**Reply by dcbaker** [reply]
That means it's not finding it. Can you post the full output of `meson setup $builddir`?

**Reply by ghost** [reply]
Ah, ok. Here's the output:
```
:/opt/mellanox/doca/applications$ meson $/tmp/build
The Meson build system
Version: 0.61.2
Source dir: /opt/mellanox/doca/applications
Build dir: /tmp/build
Build type: native build
Program cat found: YES (/usr/bin/cat)
Project name: DOCA_APPLICATIONS
Project version: 2.2.0080
C compiler for the host machine: cc (gcc 11.4.0 "cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0")
C linker for the host machine: cc ld.bfd 2.38
C++ compiler for the host machine: c++ (gcc 11.4.0 "c++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0")
C++ linker for the host machine: c++ ld.bfd 2.38
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program /usr/local/cuda-12.2/bin/nvcc found: YES (/usr/local/cuda-12.2/bin/nvcc)
meson.build:36: WARNING: Module unstable-cuda has no backwards or forwards compatibility and might not exist in future releases.
Run-time dependency CUDA (modules: cudart, cuda) found: YES 12.2 (/usr/local/cuda)
Run-time dependency threads found: YES
Found pkg-config: /usr/bin/pkg-config (0.29.2)
Run-time dependency json-c found: YES 0.15
Run-time dependency libbsd found: YES 0.11.5
Program /usr/local/cuda-12.2/bin/nvcc found: YES (/usr/local/cuda-12.2/bin/nvcc)
Dependency cuda (modules: cudart, cuda) found: YES 12.2 (cached)
Run-time dependency doca found: YES 2.2.0080
Run-time dependency libdpdk found: YES 22.11.2307.2.0
Run-time dependency doca-gpu found: YES 2.2.0080

gpu_packet_processing/src/meson.build:70:0: ERROR: No host machine compiler for "gpu_packet_processing/src/gpu_kernels/receive_tcp.cu"
```

**Reply by dcbaker** [reply]
That doesn't look right, you should have a compiler for cuda, either from `project('doca_application', 'c', 'c++', 'cuda')` or `add_language('cuda')`. it should give a print out similar to the one for C++ above like:
```
Cuda compiler for the host machine: ...
Cuda linker for the host machine: ...
```

**Reply by ghost** [reply, marked as answer]
Hey, thanks a lot. Turns out the meson.build file that came with DOCA was improperly configured. I had tried add_language('cuda') but it didn't work. But adding 'cuda' to project() did the trick.
